### PR TITLE
Change some conditions from equal to identical

### PR DESCRIPTION
**Pull request by cheprasov**
| Q | A |
| --- | --- |
| Branch? | "master" |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

In PHP any string, that does not started from number is equal (==) 0.
Example:

``` php
var_dump(0 == '+'); // true
var_dump(0 == '*'); // true
var_dump(0 == '&'); // true
var_dump(0 == 'foo'); // true

var_dump(0 != '&'); // false
```

To avoid this behavior is better to use identical operator (===)
Also, === is faster than ==

Also, for **[HttpFoundation / Request]** I make logic easier

from

```
(false !== $pos = strpos($path, $baseUrl)) && 0 != $pos)
// result is not false and is not 0
```

to

```
strpos($path, $baseUrl)
// it is mean, result is true
```

PS helps to avoid some bugs in future.

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php [hunk 1]
```diff
@@ -202,13 +202,13 @@ protected function registerMappingDrivers($objectManager, ContainerBuilder $cont
             if ($container->hasDefinition($mappingService)) {
                 $mappingDriverDef = $container->getDefinition($mappingService);
                 $args = $mappingDriverDef->getArguments();
-                if ($driverType == 'annotation') {
+                if ($driverType === 'annotation') {
```
Comment (Simperfit) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: You can go for yoda style

File: src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php [hunk 2]
```diff
@@ -202,13 +202,13 @@ protected function registerMappingDrivers($objectManager, ContainerBuilder $cont
...
                     $args[1] = array_merge(array_values($driverPaths), $args[1]);
                 } else {
                     $args[0] = array_merge(array_values($driverPaths), $args[0]);
                 }
                 $mappingDriverDef->setArguments($args);
-            } elseif ($driverType == 'annotation') {
+            } elseif ($driverType === 'annotation') {
```
Comment (Simperfit) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: same in here

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php [hunk 3]
```diff
@@ -177,7 +177,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
                 $extractedMessagesCount += $domainMessagesCount;
             }
 
-            if ($input->getOption('output-format') == 'xlf') {
+            if ($input->getOption('output-format') === 'xlf') {
```
Comment (Simperfit) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: Yoda style ?

**Comment by Simperfit** [comment]
Could you please change to Yoda style when it is needed ?

**Comment by cheprasov** [comment]
Hello Simperfit. Thanks for the comments. 
Let's leave a change to Yoda style for another PR, that will do it for whole project.

**Comment by nicolas-grekas** [comment]
We can't accept a huge PR like this. We already use strict comparisons most of the time. This is enforced only by review, which means we can miss them sometimes as you spotted. Yet, once accepted, the potential for BC break is real and the time for reviewing is high but the practical benefit is low, which means it's not worth it. Btw, for smaller PR, this should be submitted against 2.7, where such clean ups are applied first. For all these reasons, I'm closing. Thanks for submitting anyway!

**Pull request closed without merging**